deploy/scripts/init_certs.sh

```diff
@@ -13,6 +13,85 @@ else
     exit 0
 fi
 
+# Get system hostnames
+hostname_short=$(hostname)
+hostname_fqdn=$(hostname -f || hostname)
+
+# Get all host IP addresses
+mapfile -t host_ips < <(hostname -I | tr ' ' '\n' | grep -v '^$')
+
+# Validate the base domain from config, or default to hostname FQDN
+base_domain=$(python3 "$YAML_HELPER" extract .global.baseDomain "$CONFIG_FILE")
+if [[ -z "$base_domain" ]]; then
+    base_domain="$hostname_fqdn"
+    echo "global.baseDomain not set, defaulting to system hostname FQDN ($base_domain)"
+fi
+
+# Validate as hostname or FQDN: lowercase alphanumerics and hyphens, final label must start with letter
+if ! [[ "$base_domain" =~ ^([a-z0-9]([-a-z0-9]*[a-z0-9])?\.)*[a-z]([-a-z0-9]*[a-z0-9])?$ ]]; then
+    echo "ERROR: global.baseDomain must be a valid hostname or FQDN (not an IP address)" 1>&2
+    exit 1
+fi
+
+# Build SAN arrays for each certificate type
+# SANs include:
+#  * External DNS names (based on baseDomain)
+#  * System hostnames (short and FQDN)
+#  * Podman service names
+#  * Loopback address
+#  * All host IP addresses
+
+# API certificate SANs
+api_sans=(
+    "api.$base_domain"
+    "agent-api.$base_domain"
+    "$base_domain"
+    "$hostname_short"
+    "$hostname_fqdn"
+    "flightctl-api"
+    "localhost"
+    "127.0.0.1"
```

Comment:
there are security implications to allowing localhost in the SAN I suppose this is a non production self signed cert.

ref, https://letsencrypt.org/docs/certificates-for-localhost/